namui/src/namui/manager/mod.rs

```diff
@@ -16,10 +17,24 @@ pub mod mock;
 pub use mock::*;
 
 pub struct Managers {
-    pub mouse_manager: Box<MouseManager>,
-    pub font_manager: Box<FontManager>,
-    pub keyboard_manager: Box<KeyboardManager>,
-    pub screen_manager: Box<ScreenManager>,
-    pub image_manager: Arc<ImageManager>,
-    pub wheel_manager: Box<WheelManager>,
+    pub mouse_manager: MouseManager,
+    pub font_manager: FontManager,
+    pub keyboard_manager: KeyboardManager,
+    pub screen_manager: ScreenManager,
+    pub image_manager: ImageManager,
+    pub wheel_manager: WheelManager,
+}
+
+pub(crate) static MANAGERS: OnceCell<ReentrantMutex<Arc<Managers>>> = OnceCell::new();
+pub fn managers() -> ReentrantMutexGuard<'static, Arc<Managers>> {
+    MANAGERS.get().expect("managers not initialized").lock()
+}
+
+pub(crate) fn set_managers(managers: Managers) {
+    if MANAGERS
+        .set(ReentrantMutex::new(Arc::new(managers)))
+        .is_err()
+    {
+        panic!("managers already initialized");
+    }
```

Comment:
This is the core of this PR.
1. Use Reentrant Mutex for `fn managers()`
2. Remove all Boxing of Managers inside
3. manager's pub fn should use immutable reference